test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,597 @@
+                name: "modWithADependency",
+                dependencies: ["author-mod-1.0.0"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-mod-2.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-mod-1.0.0"]}
+                ]
+            });
+            const combos = [comboWithADependency];
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("A mod pack with a dependency", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-modPack-2.0.0", "author-mod-1.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-modPack-2.0.0", "author-mod-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const mod = addMockPackage({
+                name: "mod",
+                versions: [
+                    {version: "2.0.0"},
+                    {version: "1.0.0"}
+                ]
+            });
+            const modPack = addMockPackage({
+                name: "modPack",
+                dependencies: ["author-mod-1.0.0"],
+                categories: ["Modpacks"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-mod-1.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-mod-1.0.0"]}
+                ]
+            });
+
+            const combos = [modPack];
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("A mod pack with a dependency, with a newer version of the dependency already installed", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-modPack-2.0.0", "author-mod-1.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-modPack-2.0.0", "author-mod-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const mod = addMockPackage({
+                name: "mod",
+                versions: [
+                    {version: "2.0.0"},
+                    {version: "1.0.0"}
+                ]
+            });
+            const modPack = addMockPackage({
+                name: "modPack",
+                dependencies: ["author-mod-1.0.0"],
+                categories: ["Modpacks"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-mod-1.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-mod-1.0.0"]}
+                ]
+            });
+
+            const installedMod = new ManifestV2().fromThunderstoreCombo(mod);
+            installedMod.setVersionNumber(new VersionNumber("2.0.0"));
+
+            const combos = [modPack];
+            expectToContainOnly(await getFullDependencyList(combos, game, [installedMod], installMode), expected);
+        });
+    });
+
+    describe("A mod pack first and a mod with the same dependency last", () => {
```

Comment:
I like that we have tests to check also combinations of `INSTALL_SPECIFIC` and passing multiple combos work in a sane fashion. At the same time they test behaviour that is not used nor formally specified. In practice I think this means the tests are now working as the spec, which would normally be fine. I'm just thinking of a scenario where someone augments `getFullDependencyList` when they actually want to install multiple specific mods, and then the tests break. In such case, I would be hesitant to move forward with the changes. I was wondering if we should alleviate those fears with a comment or something?

Just thinking out loud, not asking for any changes at the moment.

Reply:
I added a comment (L67->), which should clarify it at least a little bit.